parcels/field.py

```diff
@@ -33,6 +35,11 @@
 )
 from .grid import CGrid, Grid, GridType
 
+if TYPE_CHECKING:
+    from ctypes import _Pointer as PointerType
+
+    from parcels.fieldset import FieldSet
+
```

Comment:
Missed this in our meeting. The `TYPE_CHECKING` boolean is set by `mypy` (and also editors like vscode) when it goes through to evaluate types. This allows you to expose things to mypy and type checkers which has some benefits:

- Avoiding circular imports (which is why its done here, `parcels.fieldset` imports `Field`, so if we have a `from parcels.fieldset import FieldSet` here we run into a circular import)
- Reducing runtime overhead slightly by avoiding an import if its only used for type hinting
- to set arbitrary custom behaviour when typechecking

Note that when this is done you have to use the string `"FieldSet"` instead of `FieldSet` as a type hint in order to avoid errors at runtime (but the type checking still works out)